common:
    lidar_topic:  "/livox/lidar"
    imu_topic:  "/livox/imu"
    map_frame:  "camera_init"
    body_frame:  "body"
    timedelay_lidar2imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).

preprocess:
    lidar_type: 1                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 6
    blind: 0.1
    det_range: 450.0

mapping:
    start_in_aggressive_motion: false   # if true, a preknown gravity should be provided in following gravity_init
    gravity_init: [0, 0, -9.81]         # [0.0, 9.810, 0.0] # preknown gravity in the first IMU body frame, use when imu_en is false or start from a non-stationary state
    preset_gravity: [0, 0, -9.81]       # [0.0, 9.810, 0.0] # the direction of gravity(map) to be aligned
    gravity_align: true                 # true to align the z axis of world frame with the direction of gravity, and the gravity direction should be specified below
    frontend_type: 0                    # frontend type

    # fastlio
    point_skip_num: 4
    space_down_sample: true
    surf_frame_ds_res: 0.5
    max_iteration: 3
    ikdtree_resolution: 0.5
    lidar_model_search_range: 5
    lidar_meas_cov: 0.001
    cube_side_length: 1000
    runtime_log_enable: 0

    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic
    # extrinsic_T: [ 0.04165, 0.02326, -0.0284 ]  # avia
    extrinsic_T: [ -0.011, -0.02329, 0.04412 ]  # mid360
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]

scan_context:
    lidar_height: 2
    sc_dist_thres: 0.5

publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.

official:
    save_globalmap_en: false
    save_resolution: 0.1
    save_keyframe_en: false
